src/Agent.Worker/Build/GitSourceProvider.cs

```diff
@@ -300,18 +300,19 @@ public abstract class GitSourceProvider : SourceProvider, ISourceProvider
             Trace.Info($"gitLfsSupport={gitLfsSupport}");
             Trace.Info($"acceptUntrustedCerts={acceptUntrustedCerts}");
 
+            bool preferGitFromPath;
 #if OS_WINDOWS
             bool schannelSslBackend = HostContext.GetService<IConfigurationStore>().GetAgentRuntimeOptions()?.GitUseSecureChannel ?? false;
             Trace.Info($"schannelSslBackend={schannelSslBackend}");
-#endif
-
+            
             // Determine which git will be use
             // On windows, we prefer the built-in portable git within the agent's externals folder, set system.prefergitfrompath=true can change the behavior, 
             // agent will find git.exe from %PATH%
             // On Linux, we will always use git find in %PATH% regardless of system.prefergitfrompath
-            bool preferGitFromPath = executionContext.Variables.GetBoolean(Constants.Variables.System.PreferGitFromPath) ?? false;
-#if !OS_WINDOWS
-            // On linux/OSX, we will always find git from %PATH%
+            bool overrideGitFromPath;
```

Comment:
`bool overrideGitFromPath;` is duplicate with `bool preferGitFromPath;`
context variable should be able to override env variable.
maybe something like this would be better
```C#
preferGitFromPath = executionContext.Variables.GetBoolean(Constants.Variables.System.PreferGitFromPath) ?? false;
if(!preferGitFromPath)
{
  bool.TryParse(Environment.GetEnvironmentVariable(Constants.Variables.System.PreferGitFromPath), out preferGitFromPath);
}
```